scripts/shtf.py

```diff
@@ -0,0 +1,629 @@
+import json
+import logging
+import os
+import time
+from collections import defaultdict
+from dataclasses import dataclass, field
+from datetime import datetime
+from pathlib import Path
+from typing import Callable, DefaultDict, Optional, TypedDict, cast
+
+import click
+import requests
+import toml
+from eth_typing import BlockNumber
+from typing_extensions import NotRequired
+from web3 import Web3
+from web3.constants import ADDRESS_ZERO
+from web3.middleware import geth_poa_middleware
+from web3.types import BlockData
+
+import beamer.contracts
+import beamer.events
+from beamer.config import _merge_dicts
+from beamer.events import ClaimMade, DepositWithdrawn, RequestCreated, RequestFilled
+from beamer.typing import ChainId
+from beamer.util import load_ERC20_abi, setup_logging
+
+logging.disable()
+
+
+class MessageTypes:
+    REQUEST_EXPIRED = "RequestExpired"
+    DEPOSIT_WITHDRAWN = "DepositWithdrawn"
+    UNCLAIMED_FILL = "UnclaimedFill"
+    CHALLENGE_GAME = "ChallengeGame"
+    CHALLENGE_GAME_CLAIM_BY_SOMEONE_ELSE = "ChallengeGameClaimedBySomeoneElse"
+
+
+class MessageType(TypedDict):
+    text: str
+    message_link: NotRequired[str]
+
+
+TokensConfig = dict[str, list[tuple[str, str]]]
+
+
+class ConfigType(TypedDict):
+    agent_address: str
+    deployment_dir: Path
+    notification_system: str
+    rpcs: dict[int, tuple[str, BlockNumber]]
+    explorers: dict[int, str]
+    notification: dict[str, dict[str, str]]
+    tokens: TokensConfig
+
+
+class TransferType(TypedDict):
```

Comment:
I didnt really get this actually. Name implies that this is just types like an enum, if not maybe we can just rename it to TransferInfo or something

Reply:
I would actually prefer to have call it just Transfer, but python doesn't seem to have a way to indicate that this is a type - other than adding this to the name?

In typescript I could do
type Transfer = ...
or 
interface Transfer

A beamer Transfer is composed of a created, filled, claimed and withdrawn events. If you want we can call this TransferInfo, but I don't see it as info, but as an actual Transfer :D

Reply:
Like i wrote to you in rocketchat, we may have RequestEventHistory as dataclass for this purpose.